tests/compile-fail/never_say_never.rs

```diff
@@ -4,7 +4,7 @@
 fn main() {
     let y = &5;
     let x: ! = unsafe {
-        *(y as *const _ as *const !) //~ ERROR entered unreachable code
+        *(y as *const _ as *const !) //~ ERROR tried to access a dead local variable
```

Comment:
Here's the MIR for main:
```
fn main() -> () {
    let mut _0: ();                      // return pointer
    scope 1 {
        let _1: &i32;                    // "y" in scope 1 at src/main.rs:5:9: 5:10
        scope 2 {
            let _2: !;                   // "x" in scope 2 at src/main.rs:6:9: 6:10
        }
    }
    let mut _3: !;
    let mut _4: *const !;
    let mut _5: *const i32;
    let mut _6: *const i32;
    let mut _7: &i32;
    let mut _8: !;

    bb0: {
        StorageLive(_1);                 // scope 0 at src/main.rs:5:9: 5:10
        _1 = promoted0;                  // scope 0 at src/main.rs:5:13: 5:15
        StorageLive(_2);                 // scope 1 at src/main.rs:6:9: 6:10
        StorageLive(_4);                 // scope 1 at src/main.rs:7:10: 7:37
        StorageLive(_5);                 // scope 1 at src/main.rs:7:11: 7:24
        StorageLive(_6);                 // scope 1 at src/main.rs:7:11: 7:12
        StorageLive(_7);                 // scope 1 at src/main.rs:7:11: 7:12
        _7 = _1;                         // scope 1 at src/main.rs:7:11: 7:12
        _6 = _7 as *const i32 (Misc);    // scope 1 at src/main.rs:7:11: 7:12
        StorageDead(_7);                 // scope 1 at src/main.rs:7:12: 7:12
        _5 = _6;                         // scope 1 at src/main.rs:7:11: 7:24
        StorageDead(_6);                 // scope 1 at src/main.rs:7:24: 7:24
        _4 = _5 as *const ! (Misc);      // scope 1 at src/main.rs:7:10: 7:37
        StorageDead(_5);                 // scope 1 at src/main.rs:7:37: 7:37
        _3 = (*_4);                      // scope 1 at src/main.rs:7:9: 7:37
        _2 = _3;                         // scope 1 at src/main.rs:7:9: 7:37
        StorageDead(_3);                 // scope 1 at src/main.rs:7:37: 7:37
        StorageDead(_4);                 // scope 1 at src/main.rs:8:7: 8:7
        _8 = _2;                         // scope 2 at src/main.rs:9:7: 9:8
        const f(_8);                     // scope 2 at src/main.rs:9:5: 9:9
    }
}
```
Notice that _3 is never marked live. It hence remains dead, which makes the write to it invalid.